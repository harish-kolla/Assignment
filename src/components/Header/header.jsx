import React from "react";
import Button from "react-bootstrap/Button";

import { staticKeys } from "../../constants";

export const HeaderComponent = ({ addNewRow, scrollToTop }) => {
  return (
    <nav
      className="navbar sticky-top navbar-light "
      style={{ background: "#e3f2fd" }}
    >
      <Button
        data-testid="add-row-button"
        variant="primary"
        className="add-buttton"
        onClick={addNewRow}
      >
        {staticKeys.addRow}
      </Button>

      <span data-testid="virtulized-list" className="virtulized-list">
        {staticKeys.virtualizedList}
      </span>
      <button
        data-testid="scroll-top-button"
        type="button"
        className="btn btn-link"
        onClick={scrollToTop}
      >
        {staticKeys.topPage}
      </button>
    </nav>
  );
};
