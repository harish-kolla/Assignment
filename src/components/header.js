import React from "react";
import Button from "react-bootstrap/Button";

export const HeaderComponent = ({ addNewRow, scrollToTop, itemsCount }) => {
  return (
    <nav
      className="navbar sticky-top navbar-light "
      style={{ background: "#e3f2fd" }}
    >
      {/* <span>
        {"Total Items: "} {itemsCount}
      </span> */}
      <Button variant="primary" className="add-buttton" onClick={addNewRow}>
        {"Add Row"}
      </Button>
      <button type="button" className="btn btn-link" onClick={scrollToTop}>
        Top Page
      </button>
      {/* <span onClick={scrollToTop}>Top Page</span> */}
    </nav>
  );
};
