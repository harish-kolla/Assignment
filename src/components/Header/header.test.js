import { render, fireEvent } from "@testing-library/react";

import { HeaderComponent } from "./header";

const testProps = {
  addNewRow: jest.fn(),
  scrollToTop: jest.fn(),
};

test("Snapshot of AddNewRecord", () => {
  const { container } = render(<HeaderComponent {...testProps} />);
  expect(container.firstChild).toMatchSnapshot();
});

test("Should render components", () => {
  const { queryByTestId } = render(<HeaderComponent {...testProps} />);
  expect(queryByTestId("add-row-button")).toBeTruthy();
  expect(queryByTestId("virtulized-list")).toBeTruthy();
  expect(queryByTestId("scroll-top-button")).toBeTruthy();
});

test("Should call addNewRow on click add row button", () => {
  const { queryByTestId } = render(<HeaderComponent {...testProps} />);
  fireEvent.click(queryByTestId("add-row-button"));
  expect(testProps.addNewRow).toHaveBeenCalledTimes(1);
});

test("Should call scrollToTop on click scroll to top button", () => {
  const { queryByTestId } = render(<HeaderComponent {...testProps} />);
  fireEvent.click(queryByTestId("scroll-top-button"));
  expect(testProps.scrollToTop).toHaveBeenCalledTimes(1);
});
