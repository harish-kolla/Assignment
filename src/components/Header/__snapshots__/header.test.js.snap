// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of AddNewRecord 1`] = `
<nav
  class="navbar sticky-top navbar-light "
  style="background: rgb(227, 242, 253);"
>
  <button
    class="add-buttton btn btn-primary"
    data-testid="add-row-button"
    type="button"
  >
    Add Row
  </button>
  <span
    class="virtulized-list"
    data-testid="virtulized-list"
  >
    Virtualized List
  </span>
  <button
    class="btn btn-link"
    data-testid="scroll-top-button"
    type="button"
  >
    Top Page
  </button>
</nav>
`;
